Readme.md

```diff
@@ -1,23 +1,47 @@
 [![](https://img.shields.io/badge/📖_How_to_use_DevExpress_Examples-e9f6fc?style=flat-square)](https://docs.devexpress.com/GeneralInformation/403183)
 [![](https://img.shields.io/badge/💬_Leave_Feedback-feecdd?style=flat-square)](#does-this-example-address-your-development-requirementsobjectives)
 <!-- default badges end -->
-# How to: Display a List of Non-Persistent Objects
+# How to: Display a List of Non-Persistent Objects via an Intermediate Container Class and Its DetailView  
 
+This example stores a list of books. When a user clicks the **Show Duplicate Books** action, a popup dialog displays a list of duplicated books and their copy counts. This list is shown with a detail view.
 
-<p>This example demonstrates how to display a list of non-persistent objects. The complete description is available in the <a href="http://help.devexpress.com/#Xaf/CustomDocument3167"><u>How to: Display a List of Non-Persistent Objects</u></a> help topic.<br />
-  
-![image](https://user-images.githubusercontent.com/14300209/232475656-ba30a5be-4e96-49f9-8228-2091378e5f09.png)
+> **Note**:
+> This example uses an intermediate container class to display a list non-persistent objects. In this case, the [ObjectsGetting](https://docs.devexpress.com/eXpressAppFramework/DevExpress.ExpressApp.NonPersistentObjectSpace.ObjectsGetting) method cannot be applied. If your application logic does not require complex operations to create non-persistent objects, use the [ObjectsGetting](https://docs.devexpress.com/eXpressAppFramework/DevExpress.ExpressApp.NonPersistentObjectSpace.ObjectsGetting) method. Refer to the following article for implementation details: [How to: Display a Non-Persistent Object's List View from the Navigation](https://docs.devexpress.com/eXpressAppFramework/114052/business-model-design-orm/non-persistent-objects/how-to-display-a-non-persistent-objects-list-view-from-the-navigation).
 
-<strong>See Also:</strong><br />
-<a href="https://www.devexpress.com/Support/Center/p/K18117">K18117</a><br />
-<a href="https://www.devexpress.com/Support/Center/p/K18083">K18083</a><br />
-<a href="https://www.devexpress.com/Support/Center/p/E911">E911</a></p>
+![Blazor application: Non-persistent objects in a dialog popup](xaf-blazor-show-npo-in-popup.png)
 
-<br/>
+## Implementation Details
 
+1. Declare the `Book` persistent class. Objects of this class represent books in a collection.
+1. Declare two non-persistent classes—`Duplicate` to store unique book titles and the total number of books with these titles; and `DuplicatesList` to aggregate the `Duplicate` objects.
+1. In a controller, implement a method that iterates through persistent `Book` objects, counts the number of copies of each book, and saves this information in a dictionary.
+1. Implement a method that iterates through dictionary items and creates `Duplicate` objects for items with a value greater than one (books with more than one copy). Arrange `Duplicate` objects into a `DuplicatesList` object.
```

Comment:
``Arrange `Duplicate` objects into a `DuplicatesList` object.``

What do you mean by "arrange" in this context? Are the `Duplicate` objects already in the `DuplicatesList` object you just need to arrange/rearrange them, or do you mean "Add" these objects to the `DuplicatesList` object?

Reply:
```suggestion
1. Implement a method that iterates through dictionary items and creates `Duplicate` objects for items with a value greater than one (books with more than one copy). Add `Duplicate` objects into a `DuplicatesList` collection.
```

Reply:
```suggestion
1. Implement a method that iterates through dictionary items and creates `Duplicate` objects for items with a value greater than one (books with more than one copy). Add `Duplicate` objects to a `DuplicatesList` collection.
```